api_app/core/credentials.py

```diff
@@ -0,0 +1,41 @@
+from contextlib import asynccontextmanager
+from core import config
+from azure.core.credentials import TokenCredential
+
+from azure.identity import (
+    DefaultAzureCredential,
+    ManagedIdentityCredential,
+    ChainedTokenCredential,
+)
+from azure.identity.aio import (
+    DefaultAzureCredential as DefaultAzureCredentialASync,
+    ManagedIdentityCredential as ManagedIdentityCredentialASync,
+    ChainedTokenCredential as ChainedTokenCredentialASync,
+)
+
+
+def get_credential() -> TokenCredential:
+    managed_identity = config.MANAGED_IDENTITY_CLIENT_ID
+    if managed_identity:
```

Comment:
maybe it's worth logging whether we use the managed identity or not

Reply:
Is it?
On Azure it will always be managed, so this is about a local run.